tests: Add tests for Org API

R2ZER0 opened the pull request:
Given the API is approaching a fairly minimal but hopefully stable core/1.0, I think this is a good time to add tests.

This PR adds tests to check all endpoints (OrgList, OrgDetail, OrgGrantsMade, OrgGrantsReceived) and all Org types (Funder, Recipient, Publisher).

R2ZER0 committed:
tests: Add tests for Org API

michaelwood commented:
Todo make sure any complex queries also get tested / avoid any potentially duplicated testing of frameworks

R2ZER0 committed:
api: Raise 404 on GrantsView if Org doesn't exist

R2ZER0 committed:
tests: Add test for Organisation.exists

R2ZER0 committed:
tests: Fix org_api recipient grants test

Fix a bug in this test missed due toprevious GrantsMadeView behaviour of returning empty list instead of 404.

R2ZER0 committed:
tests: Add Org API tests for non-existent Org behaviour

R2ZER0 committed:
api: Fix bug in OrgDetail for non-existent Org

The OrgDetail view wasn't properly handling object-not-found errors.

R2ZER0 committed:
tests: Fix example OrgIDs to be valid OrgIDs

Example OrgIDs should start with prefix XE-EXAMPLE

R2ZER0 committed:
api: Use absolute imports

Replace relative imports with absolute imports

R2ZER0 committed:
tests: Add tests for Organisation.get()
